allennlp/training/exponential_moving_average.py

```diff
@@ -0,0 +1,64 @@
+from typing import Dict, Iterable
+from allennlp.models.model import Model
+
+
+class ExponentialMovingAverage:
+    """
+    Create shadow variables and maintain exponential moving average for model parameters.
+
+    Parameters
+    ----------
+    model: ``Model``, required.
+        An AllenNLP model to be optimized.
```

Comment:
I am not convinced about this API. Why not just pass in the named parameters in the constructor (instead of the model) and then you can get rid of that option in the method calls.